twoaxistracking/tests/test_layout.py

```diff
@@ -0,0 +1,154 @@
+from twoaxistracking import layout
+from shapely import geometry
+import numpy as np
+import pytest
+
+
+@pytest.fixture
+def rectangular_geometry():
+    collector_geometry = geometry.box(-2, -1, 2, 1)
+    total_collector_area = collector_geometry.area
+    min_tracker_spacing = layout._calculate_min_tracker_spacing(collector_geometry)
+    return collector_geometry, total_collector_area, min_tracker_spacing
+
+
+@pytest.fixture
+def square_field_layout():
+    # Corresponds to GCR 0.125 with the rectangular_geometry
+    X = np.array([-8, 0, 8, -8, 8, -8, 0, 8])
+    Y = np.array([-8, -8, -8, 0, 0, 8, 8, 8])
+    tracker_distance = (X**2 + Y**2)**0.5
+    relative_azimuth = np.array([225, 180, 135, 270, 90, 315, 0, 45])
+    Z = np.zeros(8)
+    relative_slope = np.zeros(8)
+    return X, Y, Z, tracker_distance, relative_azimuth, relative_slope
+
+
+def test_min_tracker_spacing_rectangle(rectangular_geometry):
+    # Test calculation of min_tracker_spacing for a rectangular collector
+    min_tracker_spacing = layout._calculate_min_tracker_spacing(rectangular_geometry[0])
+    assert min_tracker_spacing == np.sqrt(4**2+2**2)
+
+
+def test_min_tracker_spacing_circle():
+    # Test calculation of min_tracker_spacing for a circular collector with radius 1
+    collector_geometry = geometry.Point(0, 0).buffer(1)
+    min_tracker_spacing = layout._calculate_min_tracker_spacing(collector_geometry)
+    assert min_tracker_spacing == 2
+
+
+def test_min_tracker_spacing_circle_offcenter():
+    # Test calculation of min_tracker_spacing for a circular collector with radius 1 rotating
+    # off-center around the point (0, 1)
+    collector_geometry = geometry.Point(0, 1).buffer(1)
+    min_tracker_spacing = layout._calculate_min_tracker_spacing(collector_geometry)
+    assert min_tracker_spacing == 4
+
+
+def test_min_tracker_spacingpolygon():
+    # Test calculation of min_tracker_spacing for a polygon
+    collector_geometry = geometry.Polygon([(-1, -1), (3, 2), (4, 4), (1, 2), (-1, -1)])
+    min_tracker_spacing = layout._calculate_min_tracker_spacing(collector_geometry)
+    assert min_tracker_spacing == 2 * np.sqrt(4**2 + 4**2)
```

Comment:
Is it ok with direct checking of zero and one? For example:

    assert shaded_fraction == 0

Reply:
Also, is ``np.isclose`` sufficient for the above test?

Reply:
Good point, I'll amend my previous statement to say "...a bit risky (expect for some special floating point values like zero)".  Yeah I'd say using `==` is the best option for checking `0` or `1`.

In this case I'd probably still use a dedicated assertion function over `assert function_that_returns_bool()` though.  `assert bool` hardly has any info to work with, so it can't tell you *why* the assert failed (although pytest has some magic that lets it inspect the original expression sometimes), while a dedicated assertion function has access to everything and can make a nice message.